core/src/main/java/org/arquillian/smart/testing/configuration/Configuration.java

```diff
@@ -0,0 +1,279 @@
+        try(FileReader fileReader = new FileReader(configFile)) {
+            final Yaml yaml = new Yaml();
+            return yaml.loadAs(fileReader, Configuration.class);
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to load configuration from file " + configFile.getPath(), e);
+        }
+    }
+
+    public Path dump() {
+        final LocalStorageFileAction configFile = new LocalStorage(Paths.get("").toFile())
+            .duringExecution()
+            .temporary()
+            .file(Configuration.SMART_TESTING_YML);
+        try {
+            configFile.create();
+        } catch (IOException e) {
+            throw new RuntimeException("Cannot create " + configFile.getPath() + " file", e);
+        }
+
+        try (FileWriter fileWriter = new FileWriter(configFile.getFile())) {
+            Yaml yaml = new Yaml();
+            yaml.dump(this, fileWriter);
+
+            return configFile.getPath();
+        } catch (IOException e) {
+            throw new RuntimeException("Failed to store configuration in file " + configFile.getPath(), e);
+        }
+    }
+
+    // testing
+    public static Configuration load(Path path) {
+        try (InputStream io = Files.newInputStream(path)) {
+            final Yaml yaml = new Yaml();
+            Map<String, Object> yamlConfiguration = (Map<String, Object>) yaml.load(io);
+            return parseConfiguration(yamlConfiguration);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private static Path getConfigurationFilePath(File[] files) {
+        if(files.length == 1) {
+            final File configFile = files[0];
+            logger.info("Using configuration from " + configFile.getName());
+            return configFile.toPath();
+        }
+
+        logger.warn("Found multiple config files with supported names: " + SMART_TESTING_YAML + ", "  + SMART_TESTING_YML);
+        logger.warn("Using configuration from " + SMART_TESTING_YML);
+
+        return Arrays.stream(files)
+            .filter(file -> file.getName().equals(SMART_TESTING_YML))
+            .map(File::toPath)
+            .findFirst()
+            .get();
+    }
+
+    private static Configuration parseConfiguration(Map<String, Object> yamlConfiguration) {
+        final Configuration configuration = new Configuration();
+
+        configuration.mode = RunMode.valueOf(overWriteSystemProperty(yamlConfiguration, "mode", SMART_TESTING_MODE, DEFAULT_MODE).toUpperCase());
+
+        final String strategies = overWriteSystemProperty(yamlConfiguration, "strategies", SMART_TESTING, DEFAULT_STRATEGIES);
+
+        if (containsAnyStrategy(strategies)) {
+            configuration.strategies = strategies.split("\\s*,\\s*");
+        }
+
+        configuration.disable = Boolean.valueOf(overWriteSystemProperty(yamlConfiguration, "disable", SMART_TESTING_DISABLE, "false"));
```

Comment:
The default values `false` are not necessary for the `Boolean.valueOf`

Reply:
Here we are not using `false` as second parameter to `Boolean.valueof`, where as we are using it for `overwriteSystemPreoperty` method